### PR TITLE
boards: frdm_mcxc444: Add usb support

### PR DESCRIPTION
**Pull request by NeilChen93**
Support usb for NXP frdm_mcxc444 board

### CODE REVIEW

**Comment by danieldegrasse** [comment]
Please take a look at the CI error

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Review by tmon-nordic — DISMISSED** [review]

**Review by jfischer-no — DISMISSED** [review]

**Review by kartben — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
@NeilChen93 PR needs rebase to clear conflicts

**Comment by mmahadevan108** [comment]
@NeilChen93 , kindly fix merge conflicts. Also please add information on how this was tested.

**Comment by mmahadevan108** [comment]
@NeilChen93, please update the 4.1 release notes to indicate this support was added.

**Comment by kartben** [comment]
@NeilChen93 ping?

**Comment by danieldegrasse** [comment]
@NeilChen93 thanks for rebasing- there is unfortunately another merge conflict :/, it needs another rebase

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Please rebase on `main` and resolve any merge conflicts.

File: doc/releases/release-notes-4.1.rst [hunk 1]
```diff
@@ -101,6 +101,7 @@ Boards & SoC Support
 
   * All HWMv1 board name aliases which were added as deprecated in v3.7 are now removed
     (:github:`82247`).
+  * Enabled USB on NXP ``frdm_mcxc444``
```
Comment (mmahadevan108) [inline doc/releases/release-notes-4.1.rst:104]: Please remove this change as the whole release notes for 4.1 is under discussion about needs to be included.

Comment (mmahadevan108) [reply doc/releases/release-notes-4.1.rst:104]: A rebase is required on this PR. So might as well remove it.

**Review by dleach02 — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@NeilChen93 , was this tested with the new USB Next subsystem?

**Pull request merged**